packages/backend/src/income/__tests__/validation/updateIncomeValidation.test.ts

```diff
@@ -0,0 +1,162 @@
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_USERID_MISSING,
+      )
+    })
+
+    it('should throw an error if userId is not uuid', () => {
+      const fakeInvalidInput = { ...fakeValidInput, userId: 123 }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_USERID_TYPE,
+      )
+    })
+  })
+
+  describe('when accountId is invalid', () => {
+    it('should throw an error if accountId is missing', () => {
+      const fakeInvalidInput = {
+        id: fakeValidInput.id,
+        userId: fakeValidInput.userId,
+        name: fakeValidInput.name,
+      }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_ACCOUNTID_MISSING,
+      )
+    })
+
+    it('should throw an error if accountId is not uuid', () => {
+      const fakeInvalidInput = { ...fakeValidInput, accountId: 123 }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_ACCOUNTID_TYPE,
+      )
+    })
+  })
+
+  describe('when "name" is present and is invalid', () => {
+    it('should throw an error if name is not a string', () => {
+      const fakeInvalidInput = { ...fakeValidInput, name: 123 }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_NAME_MUST_BE_STRING,
+      )
+    })
+
+    it('should throw an error if name is an empty string', () => {
+      const fakeInvalidInput = { ...fakeValidInput, name: '' }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_NAME_EMPTY,
+      )
+    })
+  })
+
+  describe('when "amount" is present and is invalid', () => {
+    it('should throw an error if amount is not a number', () => {
+      const fakeInvalidInput = { ...fakeValidInput, amount: '123' }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_AMOUNT_TYPE,
+      )
+    })
+
+    it('should throw an error if amount is negative', () => {
+      const fakeInvalidInput = { ...fakeValidInput, amount: -1 }
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(ValidationError)
+      expect(() => validateUpdateIncomeInput(fakeInvalidInput)).toThrow(
+        VALIDATION_ERROR_AMOUNT_NEGATIVE,
+      )
+    })
+  })
+
+  describe('when "date" is present and is invalid', () => {
```

Comment:
Add another unit test under this describe to test that the date is in the correct format. Use the format const DATE_REGEX_YYYY_MM_DD